core/src/Spirit/System.cpp

```diff
@@ -6,171 +6,175 @@
 
 int System_Get_Index(State * state)
 {
-    return state->idx_active_image;
+    try
+    {
+        return state->idx_active_image;
+    }
+    catch( ... )
+    {
+        Utility::Handle_Exception();
+        return -1;
```

Comment:
In case of `Exception` which is handled without termination of Spirit, I think it is better to return `-1` since `0` is a legal system index.